### PR TITLE
Add integration test for dictionary race

### PR DESCRIPTION
**Pull request by bianpengyuan**
Verified that the test will fail if api sha is not updated.

fixes https://github.com/istio/istio/issues/7934

### CODE REVIEW

**Comment by bianpengyuan** [comment]
@mandarjog @kyessenov API dep updated. ptal.

File: mixer/test/client/env/setup.go [hunk 1]
```diff
@@ -139,6 +140,11 @@ func (s *TestSetup) SetStress(stress bool) {
 	s.stress = stress
 }
 
+// SetCheckDict set the checkDict flag
+func (s *TestSetup) SetCheckDict(checkDict bool) {
```
Comment (kyessenov) [inline mixer/test/client/env/setup.go]: Can you set it to `true` in at least one of mixer/test/client tests?

Comment (bianpengyuan) [reply mixer/test/client/env/setup.go]: So sorry! forgot to commit the new test files...

**Pull request closed without merging**